File: cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    // prompt user for amount of change
    float change_dollar;
    do
    {
        change_dollar = get_float("Change owed: ");
    }
    while (change_dollar <= 0);

    // covert dolars to cents
    int change_cent = round(change_dollar * 100);

    int count = 0;

    // check to use quarters
    int quarters = 25;
    while (change_cent >= 25)
    {
        change_cent -= quarters;
        count++;
    }

    // check to use dimes
    int dimes = 10;
    while (change_cent >= 10 && change_cent < 25)
    {
        change_cent -= dimes;
        count++;
    }

    // check to see nickles
    int nickles = 5;
    while (change_cent >= 5 && change_cent < 10)
    {
        change_cent -= nickles;
        count++;
    }

    // check to see pennies
    int pennies = 1;
    while (change_cent > 0 && change_cent < 5)
    {
        change_cent -= pennies;
        count++;
    }
    printf("%i\n", count);
}

File: cash.py
# make sure that the input is numeric and greater than 0 than convert it to pennies
def check_con(change):
    while True:
        try:
            change = float(change)
        except:
            change = input("Change: ")
        else:
            while float(change) <= 0:
                change = input("Change: ")
            return float(change) * 100


# get amout of change from the user
change = input("Change: ")

# make sure that the input is numeric and greater than 0 than convert it to pennies
change = check_con(change)

# declare dictionary containing change values
change_list = {"quarter": 25, "dime": 10, "nickle": 5, "pennie": 1}

# initiate a coin counter
coin_counter = 0

# check to use quarters
while change >= change_list["quarter"]:
    change -= change_list["quarter"]
    coin_counter += 1

# check to use dimes
while change >= change_list["dime"]:
    change -= change_list["dime"]
    coin_counter += 1

# check to use nickles
while change >= change_list["nickle"]:
    change -= change_list["nickle"]
    coin_counter += 1

# check to use pennies
while change >= change_list["pennie"]:
    change -= change_list["pennie"]
    coin_counter += 1

print(f"{coin_counter}")